clients/src/main/java/org/apache/kafka/common/metrics/stats/Max.java

```diff
@@ -21,8 +21,15 @@
  */
 public final class Max extends SampledStat {
 
+    private final boolean whetherCanBeNegative;
+
     public Max() {
-        super(Double.NEGATIVE_INFINITY);
+        this(false);
```

Comment:
agreed, but we should change the outer code which creates the Max instance.  Many places. Since all the metrics are now non-negative, that's why I change the default behavior that indeed breaks the backwards compatibility rule. Which one do you prefer?